src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java

```diff
@@ -1250,13 +1250,115 @@ public void testUndetectedOS() throws ParseException {
+        WorkloadInventoryReportModel workloadInventoryReportModel = (WorkloadInventoryReportModel) queryResultsRow.get("report");
+        // BasicFields
+        Assert.assertEquals("IMS vCenter", workloadInventoryReportModel.getProvider());
+        Assert.assertEquals("V2V-DC", workloadInventoryReportModel.getDatacenter());
+        Assert.assertEquals("Cluster 1", workloadInventoryReportModel.getCluster());
+        Assert.assertEquals("vm tests", workloadInventoryReportModel.getVmName());
+        Assert.assertEquals(100000001L, workloadInventoryReportModel.getDiskSpace(), 0);
+        Assert.assertEquals(4096, workloadInventoryReportModel.getMemory().intValue());
+        Assert.assertEquals(4, workloadInventoryReportModel.getCpuCores().intValue());
+        Assert.assertEquals(WorkloadInventoryReportModel.OS_NAME_DEFAULT_VALUE, workloadInventoryReportModel.getOsDescription());
+        Assert.assertEquals(WorkloadInventoryReportModel.OS_NAME_DEFAULT_VALUE, workloadInventoryReportModel.getOsName());
+        Assert.assertEquals("VMware vCenter", workloadInventoryReportModel.getProduct());
+        Assert.assertEquals("6.5", workloadInventoryReportModel.getVersion());
+        Assert.assertEquals("esx13.v2v.bos.redhat.com", workloadInventoryReportModel.getHost_name());
+        Assert.assertEquals(new SimpleDateFormat("yyyy-M-dd'T'hh:mm:ss.S").parse("2019-09-18T14:52:45.871Z"), workloadInventoryReportModel.getCreationDate());
+        // Flags
+        Set<String> flagsIMS = workloadInventoryReportModel.getFlagsIMS();
+        Assert.assertNull(flagsIMS);
+        // Targets
+        Assert.assertEquals(1, workloadInventoryReportModel.getRecommendedTargetsIMS().size());
+        Assert.assertTrue(workloadInventoryReportModel.getRecommendedTargetsIMS().contains("None"));
+        // Complexity
+        Assert.assertEquals(WorkloadInventoryReportModel.COMPLEXITY_UNKNOWN, workloadInventoryReportModel.getComplexity());
+        // Workloads
+        Assert.assertTrue(workloadInventoryReportModel.getSsaEnabled());
+    }
+
+    @Test
+    public void testNullOSFieldsUndetectedOS() throws ParseException {
+        // create a Map with the facts (i.e. Objects) you want to put in the working memory
+        Map<String, Object> facts = new HashMap<>();
+
+        //Basic Fields
+        VMWorkloadInventoryModel vmWorkloadInventoryModel = new VMWorkloadInventoryModel();
+        vmWorkloadInventoryModel.setProvider("IMS vCenter");
+        vmWorkloadInventoryModel.setDatacenter("V2V-DC");
+        vmWorkloadInventoryModel.setCluster("Cluster 1");
+        vmWorkloadInventoryModel.setVmName("vm tests");
+        vmWorkloadInventoryModel.setDiskSpace(100000001L);
+        vmWorkloadInventoryModel.setMemory(4096L);
+        vmWorkloadInventoryModel.setCpuCores(4);
+        vmWorkloadInventoryModel.setGuestOSFullName(null);
+        vmWorkloadInventoryModel.setOsProductName(null);
```

Comment:
Could you also add this integration test with the use cases of :  
* GuestOSFullName != null + OSProductName == null  
* GuestOSFullName == null + OSProductName != null

Reply:
Done